src/commands/server/restore.ts

```diff
@@ -85,10 +115,208 @@ export default class Restore extends Command {
+            throw new Error(`Backup with name '${backupCrName}' is not successful`)
+          }
+          flags['backup-cr'] = backupCrName
+          task.title = `${task.title} ${backupCrName} found`
+        },
+      },
+      {
+        title: 'Gathering information about backup...',
+        enabled: () => flags['backup-cr'],
         task: async (_ctx: any, task: any) => {
-          const backupServerConfig = getBackupServerConfiguration(flags)
-          await requestRestore(flags.chenamespace, backupServerConfig, flags['snapshot-id'])
+          const backupCrName = flags['backup-cr']
+          const kube = new KubeHelper(flags)
+          const backupCr: V1CheClusterBackup | undefined = await kube.getCustomResource(flags.chenamespace, backupCrName, CHE_CLUSTER_API_GROUP, CHE_CLUSTER_API_VERSION, CHE_CLUSTER_BACKUP_KIND_PLURAL)
+          if (!backupCr) {
+            throw new Error(`Backup CR with name '${backupCrName}' not found`)
+          }
+
+          if (!backupCr.spec.backupServerConfigRef) {
+            throw new Error(`Backup CR '${backupCrName}' missing backup server configuration reference`)
+          }
+          if (!backupCr.status || !backupCr.status.cheVersion) {
+            throw new Error(`Backup CR '${backupCrName}' missing Che version`)
+          }
+          if (!backupCr.status || !backupCr.status.snapshotId) {
+            throw new Error(`Backup CR '${backupCrName}' missing snapshot ID`)
+          }
+
+          flags.version = backupCr.status.cheVersion
+          flags['snapshot-id'] = backupCr.status.snapshotId
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Getting backup server info...',
+        task: async (ctx: any, task: any) => {
+          // Get all information about backup server and validate it where possible
+          // before redeploying operator to requested version.
+          ctx.backupServerConfig = getBackupServerConfiguration(flags)
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Detecting additional parameters...',
+        task: async (ctx: any, task: any) => {
+          const kube = new KubeHelper(flags)
+          ctx.isOpenshift = await kube.isOpenShift()
+
+          // Set defaults for some parameters if they weren't set
+          if (!flags.installer) {
+            flags.installer = ctx.isOpenshift ? 'olm' : 'operator'
+          }
+          if (flags.installer === 'olm' && !flags['olm-channel']) {
+            if (await kube.operatorSubscriptionExists(SUBSCRIPTION_NAME, DEFAULT_OPENSHIFT_OPERATORS_NS_NAME)) {
+              flags['olm-channel'] = OLM_STABLE_ALL_NAMESPACES_CHANNEL_NAME
+            } else {
+              flags['olm-channel'] = OLM_STABLE_CHANNEL_NAME
+            }
+          }
+          if (ctx.isOpenshift) {
+            flags.platform = 'openshift'
+            flags['cluster-monitoring'] = true
```

Comment:
Hmm

Reply:
It is a guess. On this step we do not know if cluster monitoring is turned on, but since it is turned on by default, I suppose so. Even when it turned off, the only difference is a role creation...